testsuite/integration-arquillian/tests/base/src/main/java/org/keycloak/testsuite/arquillian/AuthServerTestEnricher.java

```diff
@@ -322,7 +334,50 @@ public void initializeTestContext(@Observes(precedence = 2) BeforeClass event) {
         testContextProducer.set(testContext);
     }
 
-    public void initializeOAuthClient(@Observes(precedence = 3) BeforeClass event) {
+    public void initializeTLS(@Observes(precedence = 3) BeforeClass event) throws Exception {
+        // TLS for Undertow is configured in KeycloakOnUndertow since it requires
+        // SSLContext while initializing HTTPS handlers
+        if (AUTH_SERVER_SSL_REQUIRED && isAuthServerJBossBased()) {
+            log.info("\n\n### Setting up TLS ##\n\n");
+
+            try {
+                OnlineManagementClient client = getManagementClient();
+                Administration administration = new Administration(client);
+                Operations operations = new Operations(client);
+
+                if(!operations.exists(Address.coreService("management").and("security-realm", "UndertowRealm"))) {
```

Comment:
If we want to be future proof maybe we should consider Elytron configuration instead of using legacy configuration to enable TLS. Optional though, I'm OK too if you want to stick with legacy config.

Reply:
Since that piece is basically copy-pasted (and works fine on both EAP and WF), I would prefer to leave it as is. But thanks a lot for suggestion!